include/seqan3/alphabet/nucleotide/dna3bs.hpp

```diff
@@ -0,0 +1,189 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \author Sara Hetzel <sara.hetzel AT fu-berlin.de>
+ * \brief Provides seqan3::dna3bs, container aliases and string literals.
+ */
+
+#pragma once
+
+#include <vector>
+
+#include <seqan3/alphabet/nucleotide/nucleotide_base.hpp>
+#include <seqan3/core/char_operations/transform.hpp>
+
+// ------------------------------------------------------------------
+// dna3bs
+// ------------------------------------------------------------------
+
+namespace seqan3
+{
+/*!\brief The three letter reduced DNA alphabet for bisulfite sequencing mode (A,G,T(=C)).
+ * \ingroup nucleotide
+ * \implements seqan3::NucleotideAlphabet
+ * \implements seqan3::WritableAlphabet
+ * \if DEV \implements seqan3::detail::WritableConstexprAlphabet \endif
+ * \implements seqan3::TriviallyCopyable
+ * \implements seqan3::StandardLayout
+ * \implements std::Regular
+ *
+ * \details
+ * Note that you can assign 'U' as a character to dna3bs and it will silently
+ * be converted to 'T'.
+ * This alphabet represents a reduced version that can be used when dealing with bisulfite-converted data.
+ * All 'C's are converted to a 'T' in order to allow comparison of normal sequences with
+ * bisulfite-converted sequences.
+ * For completeness, this nucleotide alphabet has a complement table, however, it is not recommended to use
+ * it when dealing with bisulfite data because the complement of T is ambiguous in reads from bisulfite
+ * sequencing. A 'T' can represent a true thymidine or an unmethylated 'C' that was converted into a 'T'.
+ * Therefore, complementing a dna4bs sequence will further reduce the alphabet to only 'T' and 'A', thereby
+ * loosing all information about 'G'. It is recommended that when working with bisulfite data the reverse complement
+ * from dna4/5/15 is used prior to conversion into dna3bs in order to avoid simplifying the data by automatically
+ * setting 'A' as the complement of 'C'. As an example: The sequence 'ACGTGC' in dna4 would be 'ATGTGT' in dna3bs.
```

Comment:
The last sentence is quite difficult to read, maybe reorder it a little:

When working with bisulfite data, we recommend to create the reverse complement of the dan4/5/15 range first and convert to dna3bs later. This avoids...